scripts/listen_to_events/strategies.py

```diff
@@ -130,6 +130,21 @@ async def format_data(self, _chain, data):
 
     async def discord_channels(self):
         raise NotImplementedError("Subclasses should implement this method")
+    
+    async def get_pool_address(self):
+        raise NotImplementedError("Subclasses should implement this method")
+
+    async def check_if_pool_has_right_vault(self, chain, event):
+        web3 = Web3Provider.get_instance(chain, {}, NOTIFICATION_CHAIN_MAP)
+        pool_address = web3.to_checksum_address(await self.get_pool_address(event))
+        try:
+            pool = web3.eth.contract(address=pool_address, abi=get_mock_pool_abi())
+            pool_vault = await pool.functions.getVault().call()
+            return pool_vault == "0xBA12222222228d8Ba445958a75a0704d566BF2C8"
```

Comment:
`#would` normalize/lowercase both addresses 🤷  I never know what case these libs are going to throw as us